src/uncertainties.jl

```diff
@@ -0,0 +1,41 @@
+## New Type: Quantity
+# Struct for carrying statistic and systematic uncertainty of a quantity
+struct Quantity{T<:Real} <: Real
```

Comment:
I'm am unsure if we really want to restrict ourselves to `T<:Real`- quite a few quantities are complex on the lattice and I think it would be conceptually clearer if we have the option to treat them as such.

Reply:
Hmm that's a good point, however final observables need to be real.
And also I don't know how to handle errors properly for complex quantities?

Reply:
I guess that depends on the observable. But I guess it is mostly elementwise, since usually real and imiaginary parts contain different physical observation and don't really talk to each other?